src/components/check-your-email-security-codes/send-email-otp-middleware.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response, NextFunction } from "express";
+import { sendNotificationService } from "../common/send-notification/send-notification-service";
+import { NOTIFICATION_TYPE } from "../../app.constants";
+import { getErrorPathByCode } from "../common/constants";
+import { BadRequestError } from "../../utils/error";
+import xss from "xss";
+
+export async function sendEmailOtp(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  notificationService = sendNotificationService()
+): Promise<void> {
+  const email = req.session.user.email.toLowerCase();
+  const phoneNumber = req.session.user.phoneNumber;
+  const { sessionId, clientSessionId, persistentSessionId } = res.locals;
+
+  const sendNotificationResponse = await notificationService.sendNotification(
+    sessionId,
+    clientSessionId,
+    email,
+    NOTIFICATION_TYPE.VERIFY_CHANGE_HOW_GET_SECURITY_CODES,
+    req.ip,
+    persistentSessionId,
+    xss(req.cookies.lng as string),
+    phoneNumber,
+    false
+  );
+
+  if (sendNotificationResponse.success) {
+    return next();
+  }
+
+  let path: string | undefined;
+  if (sendNotificationResponse.data?.code) {
+    path = getErrorPathByCode(sendNotificationResponse.data.code);
+  }
+
+  if (path) {
+    return res.redirect(path);
+  }
```

Comment:
Yeah, I reckon so . Will push a change - thanks!

Reply:
Although, now I think about it more, I remember why it's this way. `sendNotificationResponse.data.code` could theoretically have a code that does not have a corresponding error path, so in this case we throw a more graceful error than we would just by `res.redirect()` with an `undefined` value for `path`. However, I'd say this would be a code bug if that ever happened, so still think your approach is more elegant.

We have this exact pattern in quite a few places in the existing code e.g.:

https://github.com/alphagov/di-authentication-frontend/blob/main/src/components/common/verify-code/verify-code-controller.ts#L47-53